File: server/src/main/java/com/genealogy/domain/family/Role.java
package com.genealogy.domain.family;

public enum Role {
    ADMIN("admin"),
    EDITOR("editor"),
    VIEWER("viewer");

    private final String value;

    Role(String value) {
        this.value = value;
    }

    public String getValue() {
        return value;
    }

    public boolean canWrite() {
        return this == ADMIN || this == EDITOR;
    }

    public boolean canRead() {
        return true;
    }
}

File: internal/domain/family/role.go
package family

type Role string

const (
	// RoleAdmin is the creator/owner role with full write access.
	// In P0, this is the only role that can write.
	RoleAdmin Role = "admin"

	// RoleEditor is reserved for future use.
	// Will have write access but cannot manage members.
	// NOT IMPLEMENTED IN P0 - no invite/transfer APIs.
	RoleEditor Role = "editor"

	// RoleViewer is reserved for future use.
	// Read-only access to family data.
	// NOT IMPLEMENTED IN P0 - no invite/transfer APIs.
	RoleViewer Role = "viewer"
)

func (r Role) IsValid() bool {
	switch r {
	case RoleAdmin, RoleEditor, RoleViewer:
		return true
	}
	return false
}

func (r Role) CanWrite() bool {
	switch r {
	case RoleAdmin, RoleEditor:
		return true
	}
	return false
}

func (r Role) CanRead() bool {
	return r.IsValid()
}
